tui/disk_util.go

```diff
@@ -0,0 +1,333 @@
+	"strings"
+	"syscall"
+
+	"github.com/VladimirMarkelov/clui"
+	"gopkg.in/yaml.v2"
+
+	"github.com/clearlinux/clr-installer/log"
+	"github.com/clearlinux/clr-installer/model"
+	"github.com/clearlinux/clr-installer/storage"
+	"github.com/clearlinux/clr-installer/utils"
+)
+
+const (
+	diskUtilTitle = `Launch Disk Utility`
+	diskUtil      = `cfdisk`
+)
+
+// DiskUtilPage is the Page implementation for the disk partitioning menu page
+type DiskUtilPage struct {
+	BasePage
+
+	descrptionLabel *clui.Label
+
+	chooserList *clui.ListBox
+
+	diskUtilTargets []storage.InstallTarget
+
+	labelWarning *clui.Label
+
+	devs         []*storage.BlockDevice
+	activeDisk   *storage.BlockDevice
+	activeSerial string
+}
+
+// GetConfiguredValue Returns the string representation of currently value set
+func (page *DiskUtilPage) GetConfiguredValue() string {
+	return ""
+}
+
+// GetConfigDefinition returns if the config was interactively defined by the user,
+// was loaded from a config file or if the config is not set.
+func (page *DiskUtilPage) GetConfigDefinition() int {
+	return ConfigNotDefined
+}
+
+// SetDone sets the configured disk into the model and sets the page as done
+func (page *DiskUtilPage) SetDone(done bool) bool {
+	log.Debug("%s diskUtil called for Target %v", diskUtil,
+		page.diskUtilTargets[page.chooserList.SelectedItem()])
+	page.rundiskUtil(page.diskUtilTargets[page.chooserList.SelectedItem()].Name)
+
+	// disk pages
+	page.tui.gotoPage(TuiPageMediaConfig, page)
+
+	return false
+}
+
+// Activate updates the UI elements with the most current list of block devices
+func (page *DiskUtilPage) Activate() {
+
+	page.confirmBtn.SetEnabled(false)
+
+	if err := page.buildMediaLists(); err != nil {
+		page.Panic(err)
+	}
+
+	page.activated = page.cancelBtn
+
+	clui.RefreshScreen()
+}
+
+func (page *DiskUtilPage) setConfirmButton() {
+	if page.labelWarning.Title() == "" {
+		page.confirmBtn.SetEnabled(true)
+	} else {
+		page.confirmBtn.SetEnabled(false)
+	}
+}
+
+// The disk page gives the user the option so select how to set the storage device,
+// if to manually configure it or a guided standard partition schema
+func newDiskUtilPage(tui *Tui) (Page, error) {
+	page := &DiskUtilPage{}
+	page.setup(tui, TuiPageDiskUtil, CancelButton|ConfirmButton, TuiPageMediaConfig)
+
+	// Top label for the page
+	lbl := clui.CreateLabel(page.content, 2, 2, diskUtilTitle, clui.Fixed)
+	lbl.SetPaddings(0, 1)
+
+	contentFrame := clui.CreateFrame(page.content, AutoSize, AutoSize, BorderNone, Fixed)
+	contentFrame.SetPack(clui.Vertical)
+
+	page.descrptionLabel = clui.CreateLabel(contentFrame, AutoSize, 7, "", Fixed)
+	page.descrptionLabel.SetMultiline(true)
+	page.descrptionLabel.SetPaddings(0, 2)
+	descriptiveMessage := "Launch an external program (" + diskUtil +
+		") to modify the selected\n" +
+		"disk in an attempt to allocate enough free disk space to allow\n" +
+		"the installation to succeed.\n\n" +
+		"Only disk with a 'gpt' partition can be modifify.\n\n" +
```

Comment:
Spellcheck "modify" instead of "modifify"

Reply:
fixed